src/main/io/spektrum_rssi.c

```diff
@@ -0,0 +1,200 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "platform.h"
+#ifdef USE_SERIAL_RX
+#if defined(USE_SPEKTRUM_REAL_RSSI) || defined(USE_SPEKTRUM_FAKE_RSSI)  
+
+#include "config/feature.h"
+
+#include "drivers/system.h"
+#include "drivers/time.h"
+
+#include "rx/rx.h"
+#include "rx/spektrum.h"
+#include "io/spektrum_rssi.h"
+
+
+#ifdef USE_SPEKTRUM_FAKE_RSSI
+// Spektrum Rx type. Determined by bind method.
+static bool spektrumSatInternal = true; // Assume internal,bound by BF.
+
+// Variables used for calculating a signal strength from satellite fade.
+//  This is time-variant and computed every second based on the fade
+//  count over the last second.
+static uint32_t spek_fade_last_sec = 0; // Stores the timestamp of the last second.
+static uint16_t spek_fade_last_sec_count = 0; // Stores the fade count at the last second.
+#endif
+
+// Linear mapping and interpolation function
+int32_t map(int32_t x, int32_t in_min, int32_t in_max, int32_t out_min, int32_t out_max) {
+    return (x - in_min) * (out_max - out_min) / (in_max - in_min) + out_min;
+}
+
+#ifdef USE_SPEKTRUM_RSSI_PERCENT_CONVERSION
+
+// Conversion table from dBm to a percentage scale aproximating a more linear RSSI vs Distance curve.
+
+static const stru_dbm_table dbmTable[] = {
+    {SPEKTRUM_RSSI_MAX, 101},
+    {-49,100},
+    {-56, 98},
+    {-61, 95},
+    {-66, 89},
+    {-69, 83},
+    {-71, 78},
+    {-73, 72},
+    {-74, 69},
+    {-75, 66},
+    {-76, 63},
+    {-77, 60},
+#if 0 // Linear part of the table, can be interpolated
+    {-78, 56},
+    {-79, 52},
+    {-80, 48},
+    {-81, 44},
+    {-82, 40},
+    {-83, 36},
+    {-84, 32},
+    {-85, 28},
+    {-86, 24},
+    {-87, 20}, // Beta Flight default RSSI % alatm point
+    {-88, 16},
+    {-89, 12},
+    {-90,  8}, // Failsafe usually hits here
+    {-91,  4},
+#endif // Linear part of the table, end
+    {SPEKTRUM_RSSI_MIN, 0}};
+
+#define SIZEOF_dbmTable  (sizeof(dbmTable)/sizeof(dbmTable[0]))
+
+// Convert dBm to Range %
+static int8_t dBm2range (int8_t dBm) {
+    int8_t  retval = dbmTable[0].reportAs;
+    uint8_t i = 1;
+
+    while (i < SIZEOF_dbmTable) {
```

Comment:
Use `ARRAYLEN(dbmTable)`. Also this should be a `for` loop.

Reply:
OK, fixed.